everything-cat/src/App.jsx

```diff
@@ -0,0 +1,49 @@
+import { BrowserRouter as Router, Routes, Route, Link, useLocation } from 'react-router-dom';
+import { AppBar, Toolbar, Button, Typography } from '@mui/material';
+import CatImageList from './components/dashboard/CatImageList.jsx';
+import BreedList from './components/breeds/BreedList.jsx';
+import Favourites from './components/favourites/Favourites.jsx';
+import CatImageModal from './components/modals/CatImageModal.jsx';
+
+const Navbar = () => {
+    return (
+        <AppBar position="static">
+            <Toolbar className=".MuiToolbar-root">
+                <Typography variant="h6" style={{ flexGrow: 1 }}>
+                    Cat App
+                </Typography>
+                <Button color="inherit" component={Link} to="/">Home</Button>
+                <Button color="inherit" component={Link} to="/breeds">Breeds</Button>
+                <Button color="inherit" component={Link} to="/favourites">Favourites</Button>
+            </Toolbar>
+        </AppBar>
+    );
+};
+
+const App = () => {
+    const location = useLocation();
+
+    // Check if the current path includes '/image/' to hide the navbar for the modal route
+    const isModalRoute = location.pathname.startsWith('/image/');
+
+    return (
+        <div>
+            {/* Render the navigation bar unless it's the modal route */}
+            {!isModalRoute && <Navbar />}
+            <Routes>
+                <Route path="/" element={<CatImageList />} />
+                <Route path="/image/:id" element={<CatImageModal />} />
+                <Route path="/breeds" element={<BreedList />} />
+                <Route path="/favourites" element={<Favourites />} />
+            </Routes>
+        </div>
+    );
+};
+
+const AppWrapper = () => (
+    <Router>
+        <App />
+    </Router>
+);
```

Comment:
Why not just use that in the App component instead of that "div" you used as wrapper?

Reply:
I use `const location = useLocation();` that needs to be inside a <Router> context. So I split the component in order for the  useLocation to have access to it.

Reply:
Got it. Still looks a bit odd though. I haven't followed on the latest react-router version and if it's still supported by the current API, but maybe moving `<NavBar />` inside `<Route path="/images/:id">` as child would do the trick, or even accessing window.location.path directly.

Reply:
I agree, it does feel a bit off. To achieve the same functionality using the children approach, I’d need to place `<NavBar />` inside every route except `<Route path="/images/:id">`, ensuring it appears on all pages but the modal. However, that would require repeating this for every new route, which isn't ideal for maintainability.

A better alternative would be moving `useLocation `inside `<NavBar />` to handle conditional rendering directly within the component. This would keep the App.js structure more traditional and cleaner. However, this change would turn the `<NavBar />` from a presentational to a functional component, making it more specific to this app's routing logic. As a result, it would lose reusability across other parts of the app, since it’s no longer a general-purpose component.

While this approach sacrifices some flexibility, I agree it could be worth it for simplicity in this specific case.